### PR TITLE
fix(aws/k8s): specify protocol on load balancer links based on ports/…

### PR DESCRIPTION
**Pull request by anotherchrisberry**
…listeners

If "HTTPS" is found, use that; otherwise, use "HTTP".

Tested on AWS; @lwander or @danielpeach can you try this on Kubernetes? I'm really guessing here.

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
closes https://github.com/spinnaker/spinnaker/issues/1428

**Comment by lwander** [comment]
I'll give it a shot monday, I'm away from my dev environment unfortunately.

File: app/scripts/modules/kubernetes/loadBalancer/details/details.controller.js [hunk 1]
```diff
@@ -23,22 +24,26 @@ module.exports = angular.module('spinnaker.loadBalancer.kubernetes.details.contr
...
+          $scope.loadBalancer = application.loadBalancers.data.find((test) => {
             return test.name === loadBalancer.name &&
               (test.namespace === loadBalancer.region || test.namespace === loadBalancer.namespace) &&
               test.account === loadBalancer.accountId;
-          })[0];
+          });
 
           if ($scope.loadBalancer) {
+            this.ingressProtocol = 'http:';
+            if (_.get($scope.loadBalancer, 'service.spec.ports', []).some(p => p.protocol === 'HTTPS')) {
```
Comment (lwander) [inline app/scripts/modules/kubernetes/loadBalancer/details/details.controller.js]: There is unfortunately no [`HTTPS` protocol](https://godoc.org/k8s.io/kubernetes/pkg/api#Protocol). Could you instead check if `p.port == 443`?

Comment (anotherchrisberry) [inline app/scripts/modules/kubernetes/loadBalancer/details/details.controller.js]: done

Comment (lwander) [inline app/scripts/modules/kubernetes/loadBalancer/details/details.controller.js]: thanks!

**Pull request merged**